### PR TITLE
Add tmap for terminal-job mode

### PR DESCRIPTION
**Pull request by jakalope**
This changeset adds terminal-job mode key mapping using commands `tmap`, `tnoremap`, `tunmap`, etc. Docs are updated and a test is added.

### CODE REVIEW

File: src/terminal.c [hunk 1]
```diff
@@ -1156,23 +1156,24 @@ term_enter_job_mode()
...
- * TODO: use terminal mode mappings.
+ * Use terminal mode mappings.
  */
     static int
 term_vgetc()
 {
     int c;
 
-    ++no_mapping;
+    int prev_State = State;
```
Comment (mattn) [inline src/terminal.c]: [nit] prefer save_State

File: src/testdir/test_terminal.vim [hunk 2]
```diff
@@ -627,3 +627,25 @@ func Test_terminal_redir_file()
     call delete('Xfile')
   endif
 endfunc
+
+func Test_terminal_tmap()
+  " TODO: Find an equivalent for win32
+  if !has('unix')
```
Comment (mattn) [inline src/testdir/test_terminal.vim:626]: I'll do it after merge. Thanks

**Comment by brammool** [comment]
jakalope wrote:

> This changeset adds terminal-job mode key mapping using commands
> `tmap`, `tnoremap`, `tunmap`, etc. Docs are updated and a test is
> added.

Thanks for working on this.  Can you please put your full name on your
github account?


-- 
I AM THANKFUL...
...for a lawn that needs mowing, windows that need cleaning
and gutters that need fixing because it means I have a home.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by jakalope** [comment]
@brammool Done!

File: src/terminal.c (hunk 1 above)
Comment (jakalope) [reply src/terminal.c]: Done.

**Comment by jakalope** [comment]
Several of the build configurations are failing because (I think) the systems they're running on don't recognize the `read` system command:

```
From test_terminal.vim:
Found errors in Test_terminal_wipe_buffer():
function RunTheTest[24]..Test_terminal_wipe_buffer[4]..WaitFor[21]..<SNR>5_OutCB line 3: Expected process 15540 run but got channel 32 open
function RunTheTest[24]..Test_terminal_wipe_buffer[4]..WaitFor[21]..<SNR>5_OutCB line 4: Expected 'abcde' but got 'executing job failed: No such file or directory\r\n'
```

I'm open to suggestions on other ways to test this feature.

**Comment by mattn** [comment]
I'm not sure but can you try it with `"bash -c 'read -n 4'"` ?

**Comment by jakalope** [comment]
@mattn That seems to have worked, thanks!

**Comment by jakalope** [comment]
@brammool I don't see any clear way to add tests that would (a) make sense in the context of this PR and (b) improve the test coverage. Please advise.

**Pull request closed without merging**